packages/@aws-cdk/aws-lambda/lib/code-signing-config.ts

```diff
@@ -0,0 +1,37 @@
+import { Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnCodeSigningConfig } from './lambda.generated';
+
+export enum UntrustedArtifactOnDeployment {
+  ENFORCE = 'enforce',
+  WARN = 'warn',
+}
+
+export interface CodeSigningConfigOptions {
+  signingProfileVersionArn: string[],
```

Comment:
Could you start by implementing [`SigningProfile`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-signer-signingprofile.html) as a construct in the `@aws-cdk/aws-signer` package.

The type here should be `ISigningProfile` instead of `string[]`.